macaw-ppc-symbolic/src/Data/Macaw/PPC/Symbolic/Functions.hs

```diff
@@ -117,69 +118,69 @@ funcSemantics :: (C.IsSymInterface sym, MS.ToCrucibleType mt ~ t, 1 <= SP.AddrWi
               -> S v sym rtp bs r ctx
               -> IO (C.RegValue sym t, S v sym rtp bs r ctx)
 funcSemantics sf pf s =
+  C.withBackend (s ^. C.stateContext) $ \bak ->
+  let sym = C.backendGetSym bak in
   case pf of
     MP.UDiv _rep lhs rhs -> do
-      let sym = s ^. C.stateSymInterface
-      lhs' <- toValBV sym lhs
-      rhs' <- toValBV sym rhs
+      lhs' <- toValBV bak lhs
+      rhs' <- toValBV bak rhs
       -- FIXME: Make sure that the semantics when rhs == 0 exactly match PowerPC.
       v <- LL.llvmPointer_bv sym =<< C.bvUdiv sym lhs' rhs'
       return (v, s)
     MP.SDiv _rep lhs rhs -> do
-      let sym = s ^. C.stateSymInterface
-      lhs' <- toValBV sym lhs
-      rhs' <- toValBV sym rhs
+      lhs' <- toValBV bak lhs
+      rhs' <- toValBV bak rhs
       -- FIXME: Make sure that the semantics when rhs == 0 exactly match PowerPC.
       v <- LL.llvmPointer_bv sym =<< C.bvSdiv sym lhs' rhs'
       return (v, s)
-    MP.FPNeg (_ :: MT.FloatInfoRepr fi) x -> withSymFPUnOp s x $ \sym x' ->
+    MP.FPNeg (_ :: MT.FloatInfoRepr fi) x -> withSymFPUnOp s x $ \_sym x' ->
```

Comment:
This ends up being kind of a strange pattern.  I left a bunch of `_sym` arguments in place for now to persue smaller diffs, but maybe it's worth revisiting.

Reply:
I think that is fine. Next major update we need to make to the semantics, we can revisit